src/Http/Uri.php

```diff
@@ -59,6 +60,21 @@ public function __construct(UriInterface $uri, string $base, string $webroot)
         $this->webroot = $webroot;
     }
 
+    /**
+     * Backwards compatibility shim for previously dynamic properties.
+     *
+     * @param string $name The attribute to read.
+     * @return mixed
+     */
+    public function __get(string $name)
+    {
+        if ($name === 'base' || $name === 'webroot') {
+            return $this->{$name};
+        }
+
+        throw new Error("Cannot access attribute `{$name}`");
```

Comment:
I like the notice thrown in the __get example:
```php
        $trace = debug_backtrace();
        trigger_error(
            'Undefined property via __get(): ' . $name .
            ' in ' . $trace[0]['file'] .
            ' on line ' . $trace[0]['line'],
            E_USER_NOTICE);
        return null;
```

https://www.php.net/manual/en/language.oop5.overloading.php#object.get

Reply:
Done now.